File: python/sort.py
import sys
import importlib


def main():
    try:
        algorithms = sys.argv[1]
    except IndexError:
        print "please input sort algorithms(upper) name"
        return -1

    try:
        run_num = int(sys.argv[2])
        list_len = int(sys.argv[3])
    except (IndexError, ValueError):
        run_num = 5
        list_len = 12

    name = "sort.%s" % algorithms
    try:
        package = importlib.import_module(name)
    except ImportError:
        print "not found sort algorithms: %s" % algorithms
        return -2

    object_name = '%sSort' % algorithms.title()
    if not hasattr(package, object_name):
        print "not found algorithms class: %s" % object_name
        return -3

    instance = getattr(package, object_name)
    function = getattr(instance, 'main')
    function(run_num=run_num, list_len=list_len)

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: php/sort.php
<?php
$dir = dirname(__FILE__);
$argv = $_SERVER['argv'];
if (empty($argv[1])) {
    echo "please input sort algorithm name", PHP_EOL;
    exit(-1);
}

$algorithmName = ucfirst($argv[1]);
$algorithmFile = sprintf("%s/sort/%s.php", $dir, $algorithmName);
if (!file_exists($algorithmFile)) {
    echo sprintf("%s not exists.%s", $algorithmFile, PHP_EOL);
    exit(-2);
}

require ($algorithmFile);

if (!class_exists($algorithmName)) {
    echo sprintf("class %s not found %s", $algorithmName, PHP_EOL);
    exit(-3);
}


$runNumber = empty($argv[2]) ? 1: (int) $argv[2];
$arrayLength = empty($argv[3]) ? 5: (int) $argv[3];
call_user_func_array([$algorithmName, 'main'], [$runNumber, $arrayLength]);